index.js

```diff
@@ -39,25 +33,37 @@ var lnCms = angular.module('lnCms', [
             name: route.state,
             url: route.url,
             templateUrl: 'templates/' + route.template + '/template.html',
+            controller: 'LnViewController as vm',
+            resolve: {
+              staticData: ['lnCmsClientService', function(lnCmsClientService){
+                return lnCmsClientService.getStatic();
+              }],
+              viewData: ['lnCmsClientService', '$stateParams', function(lnCmsClientService, $stateParams){
+                var endpoint = route.endpoint || 'post';
+                var params = route.params || {};
+                angular.merge(params, $stateParams);
+                return lnCmsClientService.getData(endpoint, params);
+              }],
+            },
             data: {
               endpoint: (route.endpoint || 'post'),
               fixedParams: (route.params || {}),
               request: 'request' in route ? route.request : true
             }
```

Comment:
I think that this `data` object is not needed anymore... `endpoint` and `fixedParams` are not used anywhere, and `request` is only used below in line https://github.com/moxie-lean/ng-cms/blob/e2d0cd6423ae955396fbd959784e05923a093b59/index.js#L62, but we can change that to set the `request` flag in a variable and then delete this `data` property from the state. What do you think?